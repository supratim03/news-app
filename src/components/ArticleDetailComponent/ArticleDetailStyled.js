import styled from 'styled-components'

export const ArticleContainer = styled.div`
    display: grid;
    margin-top: 12%;
    margin-left: 100px;
    margin-right: 100px;
    @media (max-width: 768px) {
        margin-top: 35%;
        margin-left: 10px;
        margin-right: 10px;
        grid-template-columns: 100%;
    }
`;

export const PublicationDateBlock = styled.div`
    padding-top: 20px
`;

export const ArticleBlock = styled.div`
    display: grid;
    grid-template-columns: 60% 40%;
    grid-gap: 20px;
    padding-top: 20px;
    @media (max-width: 768px) {
        grid-template-columns: 100%;
        grid-row-gap: 50px;
        grid-gap: 0px;
    }
`;

export const Title = styled.span`
    font-weight: bold;
    font-size: 36px;
    @media (max-width: 768px) {
        grid-template-columns: 100%;
        font-weight: bold;
        font-size: 24px;
    }
`;

export const ArticleBody = styled.div`
    padding-top: 20px;
    padding-bottom: 200px;
`;

export const BodyBlock = styled.div`
    display: grid;
    grid-template-columns: 60% 40%;
    grid-gap: 20px;
    min-height: 100vh;
    @media (max-width: 768px) {
        grid-template-columns: 100%;
        grid-row-gap: 50px;
        grid-gap: 0px;
    }
`;

export const HeadlineBlock = styled.div`
    display: grid;
    grid-template-columns: 60% 40%;
    grid-gap: 20px;
    padding-top: 20px;
    @media (max-width: 768px) {
        grid-template-columns: 100%;
        grid-row-gap: 50px;
        grid-gap: 0px;
    }
`;

export const Headline = styled.span`
    font-size: 20px;
    font-weight: 700;
    @media (max-width: 768px) {
        font-weight: 700;
        font-size: 16px;
    }
`;

export const Thumbnail = styled.div`
    background-image: url(${(props=> props.imgUrl)});
    height: 350px;
    width: auto;
    background-repeat: no-repeat;
    background-size: cover;
    @media (max-width: 768px) {
        grid-column-start: 1;
        grid-row-start: 1;
    }
`;

export const TextBlock = styled.div`
    @media (max-width: 768px) {
        grid-column-start: 1;
        grid-row-start: 2;
    }
`;