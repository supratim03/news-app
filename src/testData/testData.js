export const articleDetail = {
    "id": "music/2023/feb/11/the-trouble-with-harry-styles-triumph-at-the-brits-his-teen-fans-werent-watching",
    "type": "article",
    "sectionId": "music",
    "sectionName": "Music",
    "webPublicationDate": "2023-02-11T22:46:27Z",
    "webTitle": "The trouble with Harry Styles’ triumph at the Brits? His teen fans weren’t watching",
    "webUrl": "https://www.theguardian.com/music/2023/feb/11/the-trouble-with-harry-styles-triumph-at-the-brits-his-teen-fans-werent-watching",
    "apiUrl": "https://content.guardianapis.com/music/2023/feb/11/the-trouble-with-harry-styles-triumph-at-the-brits-his-teen-fans-werent-watching",
    "fields": {
        "headline": "The trouble with Harry Styles’ triumph at the Brits? His teen fans weren’t watching",
        "thumbnail": "https://media.guim.co.uk/5e8a1d5ab48ca3ebddf077998f7352cc6d7f1060/0_100_2991_1795/500.jpg"
    },
    "blocks": {
        "main": {
            "id": "63e815ee8f08e495c6b312ec",
            "bodyHtml": "<figure class=\"element element-image\" data-media-id=\"5e8a1d5ab48ca3ebddf077998f7352cc6d7f1060\"> <img src=\"https://media.guim.co.uk/5e8a1d5ab48ca3ebddf077998f7352cc6d7f1060/0_100_2991_1795/1000.jpg\" alt=\"Harry Styles accepting the award for artist of the year.\" width=\"1000\" height=\"600\" class=\"gu-image\" /> <figcaption> <span class=\"element-image__caption\">Harry Styles accepting the award for artist of the year.</span> <span class=\"element-image__credit\">Photograph: Dave J Hogan/Dave J. Hogan/Getty Images</span> </figcaption> </figure>",
            "bodyTextSummary": "",
            "attributes": {

            },
            "published": true,
            "createdDate": "2023-02-11T22:46:27Z",
            "firstPublishedDate": "2023-02-11T22:46:27Z",
            "publishedDate": "2023-02-11T22:46:27Z",
            "lastModifiedDate": "2023-02-11T22:26:03Z",
            "contributors": [

            ],
            "elements": [
                {
                    "type": "image",
                    "assets": [
                        {
                            "type": "image",
                            "mimeType": "image/jpeg",
                            "file": "https://media.guim.co.uk/5e8a1d5ab48ca3ebddf077998f7352cc6d7f1060/0_100_2991_1795/2000.jpg",
                            "typeData": {
                                "aspectRatio": "5:3",
                                "width": 2000,
                                "height": 1200
                            }
                        },
                        {
                            "type": "image",
                            "mimeType": "image/jpeg",
                            "file": "https://media.guim.co.uk/5e8a1d5ab48ca3ebddf077998f7352cc6d7f1060/0_100_2991_1795/1000.jpg",
                            "typeData": {
                                "aspectRatio": "5:3",
                                "width": 1000,
                                "height": 600
                            }
                        },
                        {
                            "type": "image",
                            "mimeType": "image/jpeg",
                            "file": "https://media.guim.co.uk/5e8a1d5ab48ca3ebddf077998f7352cc6d7f1060/0_100_2991_1795/500.jpg",
                            "typeData": {
                                "aspectRatio": "5:3",
                                "width": 500,
                                "height": 300
                            }
                        },
                        {
                            "type": "image",
                            "mimeType": "image/jpeg",
                            "file": "https://media.guim.co.uk/5e8a1d5ab48ca3ebddf077998f7352cc6d7f1060/0_100_2991_1795/140.jpg",
                            "typeData": {
                                "aspectRatio": "5:3",
                                "width": 140,
                                "height": 84
                            }
                        },
                        {
                            "type": "image",
                            "mimeType": "image/jpeg",
                            "file": "https://media.guim.co.uk/5e8a1d5ab48ca3ebddf077998f7352cc6d7f1060/0_100_2991_1795/2991.jpg",
                            "typeData": {
                                "aspectRatio": "5:3",
                                "width": 2991,
                                "height": 1795
                            }
                        },
                        {
                            "type": "image",
                            "mimeType": "image/jpeg",
                            "file": "https://media.guim.co.uk/5e8a1d5ab48ca3ebddf077998f7352cc6d7f1060/0_100_2991_1795/master/2991.jpg",
                            "typeData": {
                                "aspectRatio": "5:3",
                                "width": 2991,
                                "height": 1795,
                                "isMaster": true
                            }
                        }
                    ],
                    "imageTypeData": {
                        "caption": "Harry Styles accepting the award for artist of the year.",
                        "displayCredit": true,
                        "credit": "Photograph: Dave J Hogan/Dave J. Hogan/Getty Images",
                        "source": "Dave J. Hogan/Getty Images",
                        "photographer": "Dave J Hogan",
                        "alt": "Harry Styles accepting the award for artist of the year.",
                        "mediaId": "5e8a1d5ab48ca3ebddf077998f7352cc6d7f1060",
                        "mediaApiUri": "https://api.media.gutools.co.uk/images/5e8a1d5ab48ca3ebddf077998f7352cc6d7f1060",
                        "suppliersReference": "1465119574",
                        "imageType": "Photograph"
                    }
                }
            ]
        },
        "body": [
            {
                "id": "63e7b47f8f08b08d63d541da",
                "bodyHtml": "<p>Without wishing to sound hopelessly unpatriotic, the Brits is an awards ceremony that exists in the shadow of the <a href=\"https://www.theguardian.com/music/grammys\">Grammys</a>. It almost invariably takes place a week or so after the US Recording Academy <a href=\"https://www.theguardian.com/music/live/2023/feb/05/grammys-2023-harry-styles-bad-bunny-beyonce-adele-taylor-swift\">doles out its gongs</a> in a ceremony that’s bigger, more star-studded and with more impact than the Brits can ever hope for, not least because the public seem to have rather more investment in who wins. Declining interest in the Brits is something you suspect even its organisers are aware of: you can detect an urge to drum up more attention in this year’s decision to shift the ceremony from midweek to a Saturday for the first time.</p> <p>And sometimes, the US awards just foreshadow what’s going to happen at the Brits: from the moment <a href=\"https://www.theguardian.com/music/2023/feb/06/grammys-2023-if-someone-had-to-pip-beyonce-you-could-do-worse-than-harry-styles\">Harry Styles snatched the album of the year Grammy </a>from under Beyoncé’s nose, you somehow knew he was going to sweep the board in London. And so it proved.</p> <p>He went home with virtually everything bar the onstage Autocue: best artist, best album, best song and best pop/R&amp;B act. For good measure, his chief collaborator Kid Harpoon got best songwriter: there was always the chance the British Phonographic Industry gave it to him for his work on acclaimed US singer-songwriter <a href=\"https://www.theguardian.com/music/2022/jul/31/maggie-rogers-surrender-review-giving-in-to-intensity\">Maggie Rogers’ second album</a> rather than the multimillion-selling, No 1-in-27-countries <a href=\"https://www.theguardian.com/music/2022/may/16/harry-styles-harrys-house-review\">Harry’s House</a>, but you wouldn’t bank on it.</p>  <figure class=\"element element-image\" data-media-id=\"f019a4d1fe8e2e8b43d227a3d09e81a5fe1bfc95\"> <img src=\"https://media.guim.co.uk/f019a4d1fe8e2e8b43d227a3d09e81a5fe1bfc95/0_225_6733_4042/1000.jpg\" alt=\"Beyonce accepting the award for international song of the year.\" width=\"1000\" height=\"600\" class=\"gu-image\" /> <figcaption> <span class=\"element-image__caption\">Beyonce accepting the award for international song of the year.</span> <span class=\"element-image__credit\">Photograph: JMEnternational/Getty Images</span> </figcaption> </figure>  <p>And, really, fair enough. The Brits rewards success, and Styles has fairly conclusively owned the last 12 months – even before Harry’s House came out, he’d announced a US tour that entailed a staggering 10 nights at Madison Square Garden – so the chances of an upset in any of those categories was virtually nil.</p> <aside class=\"element element-rich-link element--thumbnail\"> <p> <span>Related: </span><a href=\"https://www.theguardian.com/music/2023/feb/03/gender-inequality-and-outdated-voting-metrics-are-the-brit-awards-still-hitting-the-wrong-notes\">Gender inequality and outdated voting metrics: are the Brit awards still hitting the wrong notes?</a> </p> </aside>  <p>The rest was similarly much as you might expect: <a href=\"https://www.theguardian.com/music/beyonce\">Beyoncé</a> winning the categories she was nominated in, <a href=\"https://www.theguardian.com/music/2021/aug/27/becky-hill-only-honest-on-the-weekend-review\">Becky Hill</a> held up as the best dance music has to offer, the pop-facing <a href=\"https://www.theguardian.com/music/2020/apr/18/im-not-trying-to-be-anything-im-not-rapper-aitch-on-his-rise-to-the-top\">Aitch</a> beating <a href=\"https://www.theguardian.com/music/stormzy\">Stormzy</a> – whose <a href=\"https://www.theguardian.com/music/2022/dec/04/stormzy-this-is-what-i-mean-review-intimate-downbeat-soul-baring\">most recent album</a> got a distinctly muted commercial reception – in hip-hop/grime/rap act.</p> <p>The latter may well prove controversial given that Aitch was the only white act in the category. Additionally, Beyoncé and British girl group Flo were the only artists of colour to take home prizes on the night. There were further headlines that the Brits would rather have avoided when Tom Grennan made off-colour remarks to Ellie Goulding about her breasts.</p> <p>In milder terms, you might try to suggest that indie bands <a href=\"https://www.theguardian.com/music/2022/nov/24/wet-leg-review-2022s-breakout-indie-stars-on-comically-good-form\">Wet Leg</a> and <a href=\"https://www.theguardian.com/music/2022/apr/21/fontaines-dc-skinty-fia-review-boldly-embracing-a-state-of-confusion\">Fontaines DC</a> triumphing (respectively: best new artist and best group; best international group) constituted an upset, but that feels like clutching at straws. Given the nature of the Brits, it’s certainly a little surprising that Fontaines beat K-pop titans <a href=\"https://www.theguardian.com/music/2022/sep/14/blackpink-k-pop-girl-group-born-pink-rule-music-world\">Blackpink</a>, but they’re hardly left-field unknowns: their last album entered the charts at No 1.</p>  <figure class=\"element element-image\" data-media-id=\"b472f2013d29cc88d5d6514e61b65ae7ff6542ba\"> <img src=\"https://media.guim.co.uk/b472f2013d29cc88d5d6514e61b65ae7ff6542ba/0_203_5568_3341/1000.jpg\" alt=\"Pop-facing Aitch beat Stormzy to best hip-hop/grime/rap act\" width=\"1000\" height=\"600\" class=\"gu-image\" /> <figcaption> <span class=\"element-image__caption\">Aitch beat Stormzy to best hip-hop/grime/rap act.</span> <span class=\"element-image__credit\">Photograph: David Fisher/Global/REX/Shutterstock</span> </figcaption> </figure>  <p>Save some shonky hosting from Mo Gilligan – introducing Lewis Capaldi as Sam Capaldi, and invariably relying on “gags” about alcohol consumption in between awards and performances – and a technical hitch that required showing a 2022 performance from Adele, pretty much everything went to plan. Except, one suspects, solving the Brits’ most obvious problem. We live in an age when pop’s primary audience of teens and young twentysomethings simply aren’t going to sit down and watch a two-and-a-bit-hour award ceremony on live TV: viewing doesn’t work like that any more.</p> <p>How moving the Brits to a Saturday was supposed to lure its target market away from social media and on-demand streaming – or indeed just going out with their mates – is an intriguing question. You rather got the feeling that most of this years’ viewers might have been old enough to remember the days when the Brits seemed to give <a href=\"https://www.theguardian.com/music/2022/apr/21/annie-lennox-30-greatest-songs-ranked\">Annie Lennox</a> an award every year, and that they might have spent the evening asking “who’s that?” whenever 2023’s stars appeared. Which can’t possibly be the Brits’ desired effect.</p>",
                "bodyTextSummary": "Without wishing to sound hopelessly unpatriotic, the Brits is an awards ceremony that exists in the shadow of the Grammys. It almost invariably takes place a week or so after the US Recording Academy doles out its gongs in a ceremony that’s bigger, more star-studded and with more impact than the Brits can ever hope for, not least because the public seem to have rather more investment in who wins. Declining interest in the Brits is something you suspect even its organisers are aware of: you can detect an urge to drum up more attention in this year’s decision to shift the ceremony from midweek to a Saturday for the first time. And sometimes, the US awards just foreshadow what’s going to happen at the Brits: from the moment Harry Styles snatched the album of the year Grammy from under Beyoncé’s nose, you somehow knew he was going to sweep the board in London. And so it proved. He went home with virtually everything bar the onstage Autocue: best artist, best album, best song and best pop/R&B act. For good measure, his chief collaborator Kid Harpoon got best songwriter: there was always the chance the British Phonographic Industry gave it to him for his work on acclaimed US singer-songwriter Maggie Rogers’ second album rather than the multimillion-selling, No 1-in-27-countries Harry’s House, but you wouldn’t bank on it.\nAnd, really, fair enough. The Brits rewards success, and Styles has fairly conclusively owned the last 12 months – even before Harry’s House came out, he’d announced a US tour that entailed a staggering 10 nights at Madison Square Garden – so the chances of an upset in any of those categories was virtually nil.\nThe rest was similarly much as you might expect: Beyoncé winning the categories she was nominated in, Becky Hill held up as the best dance music has to offer, the pop-facing Aitch beating Stormzy – whose most recent album got a distinctly muted commercial reception – in hip-hop/grime/rap act. The latter may well prove controversial given that Aitch was the only white act in the category. Additionally, Beyoncé and British girl group Flo were the only artists of colour to take home prizes on the night. There were further headlines that the Brits would rather have avoided when Tom Grennan made off-colour remarks to Ellie Goulding about her breasts. In milder terms, you might try to suggest that indie bands Wet Leg and Fontaines DC triumphing (respectively: best new artist and best group; best international group) constituted an upset, but that feels like clutching at straws. Given the nature of the Brits, it’s certainly a little surprising that Fontaines beat K-pop titans Blackpink, but they’re hardly left-field unknowns: their last album entered the charts at No 1.\nSave some shonky hosting from Mo Gilligan – introducing Lewis Capaldi as Sam Capaldi, and invariably relying on “gags” about alcohol consumption in between awards and performances – and a technical hitch that required showing a 2022 performance from Adele, pretty much everything went to plan. Except, one suspects, solving the Brits’ most obvious problem. We live in an age when pop’s primary audience of teens and young twentysomethings simply aren’t going to sit down and watch a two-and-a-bit-hour award ceremony on live TV: viewing doesn’t work like that any more. How moving the Brits to a Saturday was supposed to lure its target market away from social media and on-demand streaming – or indeed just going out with their mates – is an intriguing question. You rather got the feeling that most of this years’ viewers might have been old enough to remember the days when the Brits seemed to give Annie Lennox an award every year, and that they might have spent the evening asking “who’s that?” whenever 2023’s stars appeared. Which can’t possibly be the Brits’ desired effect.",
                "attributes": {

                },
                "published": true,
                "createdDate": "2023-02-11T22:46:27Z",
                "firstPublishedDate": "2023-02-11T22:46:27Z",
                "publishedDate": "2023-02-11T22:46:27Z",
                "lastModifiedDate": "2023-02-11T22:40:13Z",
                "contributors": [

                ],
                "elements": [
                    {
                        "type": "text",
                        "assets": [

                        ],
                        "textTypeData": {
                            "html": "<p>Without wishing to sound hopelessly unpatriotic, the Brits is an awards ceremony that exists in the shadow of the <a href=\"https://www.theguardian.com/music/grammys\">Grammys</a>. It almost invariably takes place a week or so after the US Recording Academy <a href=\"https://www.theguardian.com/music/live/2023/feb/05/grammys-2023-harry-styles-bad-bunny-beyonce-adele-taylor-swift\">doles out its gongs</a> in a ceremony that’s bigger, more star-studded and with more impact than the Brits can ever hope for, not least because the public seem to have rather more investment in who wins. Declining interest in the Brits is something you suspect even its organisers are aware of: you can detect an urge to drum up more attention in this year’s decision to shift the ceremony from midweek to a Saturday for the first time.</p> \n<p>And sometimes, the US awards just foreshadow what’s going to happen at the Brits: from the moment <a href=\"https://www.theguardian.com/music/2023/feb/06/grammys-2023-if-someone-had-to-pip-beyonce-you-could-do-worse-than-harry-styles\">Harry Styles snatched the album of the year Grammy </a>from under Beyoncé’s nose, you somehow knew he was going to sweep the board in London. And so it proved.</p> \n<p>He went home with virtually everything bar the onstage Autocue: best artist, best album, best song and best pop/R&amp;B act. For good measure, his chief collaborator Kid Harpoon got best songwriter: there was always the chance the British Phonographic Industry gave it to him for his work on acclaimed US singer-songwriter <a href=\"https://www.theguardian.com/music/2022/jul/31/maggie-rogers-surrender-review-giving-in-to-intensity\">Maggie Rogers’ second album</a> rather than the multimillion-selling, No 1-in-27-countries <a href=\"https://www.theguardian.com/music/2022/may/16/harry-styles-harrys-house-review\">Harry’s House</a>, but you wouldn’t bank on it.</p>"
                        }
                    },
                    {
                        "type": "image",
                        "assets": [
                            {
                                "type": "image",
                                "mimeType": "image/jpeg",
                                "file": "https://media.guim.co.uk/f019a4d1fe8e2e8b43d227a3d09e81a5fe1bfc95/0_225_6733_4042/6733.jpg",
                                "typeData": {
                                    "aspectRatio": "5:3",
                                    "width": 6733,
                                    "height": 4042
                                }
                            },
                            {
                                "type": "image",
                                "mimeType": "image/jpeg",
                                "file": "https://media.guim.co.uk/f019a4d1fe8e2e8b43d227a3d09e81a5fe1bfc95/0_225_6733_4042/master/6733.jpg",
                                "typeData": {
                                    "aspectRatio": "5:3",
                                    "width": 6733,
                                    "height": 4042,
                                    "isMaster": true
                                }
                            },
                            {
                                "type": "image",
                                "mimeType": "image/jpeg",
                                "file": "https://media.guim.co.uk/f019a4d1fe8e2e8b43d227a3d09e81a5fe1bfc95/0_225_6733_4042/2000.jpg",
                                "typeData": {
                                    "aspectRatio": "5:3",
                                    "width": 2000,
                                    "height": 1200
                                }
                            },
                            {
                                "type": "image",
                                "mimeType": "image/jpeg",
                                "file": "https://media.guim.co.uk/f019a4d1fe8e2e8b43d227a3d09e81a5fe1bfc95/0_225_6733_4042/1000.jpg",
                                "typeData": {
                                    "aspectRatio": "5:3",
                                    "width": 1000,
                                    "height": 600
                                }
                            },
                            {
                                "type": "image",
                                "mimeType": "image/jpeg",
                                "file": "https://media.guim.co.uk/f019a4d1fe8e2e8b43d227a3d09e81a5fe1bfc95/0_225_6733_4042/500.jpg",
                                "typeData": {
                                    "aspectRatio": "5:3",
                                    "width": 500,
                                    "height": 300
                                }
                            },
                            {
                                "type": "image",
                                "mimeType": "image/jpeg",
                                "file": "https://media.guim.co.uk/f019a4d1fe8e2e8b43d227a3d09e81a5fe1bfc95/0_225_6733_4042/140.jpg",
                                "typeData": {
                                    "aspectRatio": "5:3",
                                    "width": 140,
                                    "height": 84
                                }
                            }
                        ],
                        "imageTypeData": {
                            "caption": "Beyonce accepting the award for international song of the year.",
                            "displayCredit": true,
                            "credit": "Photograph: JMEnternational/Getty Images",
                            "source": "Getty Images",
                            "photographer": "JMEnternational",
                            "alt": "Beyonce accepting the award for international song of the year.",
                            "mediaId": "f019a4d1fe8e2e8b43d227a3d09e81a5fe1bfc95",
                            "mediaApiUri": "https://api.media.gutools.co.uk/images/f019a4d1fe8e2e8b43d227a3d09e81a5fe1bfc95",
                            "suppliersReference": "1465124076",
                            "imageType": "Photograph"
                        }
                    },
                    {
                        "type": "text",
                        "assets": [

                        ],
                        "textTypeData": {
                            "html": "<p>And, really, fair enough. The Brits rewards success, and Styles has fairly conclusively owned the last 12 months – even before Harry’s House came out, he’d announced a US tour that entailed a staggering 10 nights at Madison Square Garden – so the chances of an upset in any of those categories was virtually nil.</p>"
                        }
                    },
                    {
                        "type": "rich-link",
                        "assets": [

                        ],
                        "richLinkTypeData": {
                            "url": "https://www.theguardian.com/music/2023/feb/03/gender-inequality-and-outdated-voting-metrics-are-the-brit-awards-still-hitting-the-wrong-notes",
                            "originalUrl": "https://www.theguardian.com/music/2023/feb/03/gender-inequality-and-outdated-voting-metrics-are-the-brit-awards-still-hitting-the-wrong-notes",
                            "linkText": "Gender inequality and outdated voting metrics: are the Brit awards still hitting the wrong notes?",
                            "linkPrefix": "Related: ",
                            "role": "thumbnail"
                        }
                    },
                    {
                        "type": "text",
                        "assets": [

                        ],
                        "textTypeData": {
                            "html": "<p>The rest was similarly much as you might expect: <a href=\"https://www.theguardian.com/music/beyonce\">Beyoncé</a> winning the categories she was nominated in, <a href=\"https://www.theguardian.com/music/2021/aug/27/becky-hill-only-honest-on-the-weekend-review\">Becky Hill</a> held up as the best dance music has to offer, the pop-facing <a href=\"https://www.theguardian.com/music/2020/apr/18/im-not-trying-to-be-anything-im-not-rapper-aitch-on-his-rise-to-the-top\">Aitch</a> beating <a href=\"https://www.theguardian.com/music/stormzy\">Stormzy</a> – whose <a href=\"https://www.theguardian.com/music/2022/dec/04/stormzy-this-is-what-i-mean-review-intimate-downbeat-soul-baring\">most recent album</a> got a distinctly muted commercial reception – in hip-hop/grime/rap act.</p> \n<p>The latter may well prove controversial given that Aitch was the only white act in the category. Additionally, Beyoncé and British girl group Flo were the only artists of colour to take home prizes on the night. There were further headlines that the Brits would rather have avoided when Tom Grennan made off-colour remarks to Ellie Goulding about her breasts.</p> \n<p>In milder terms, you might try to suggest that indie bands <a href=\"https://www.theguardian.com/music/2022/nov/24/wet-leg-review-2022s-breakout-indie-stars-on-comically-good-form\">Wet Leg</a> and <a href=\"https://www.theguardian.com/music/2022/apr/21/fontaines-dc-skinty-fia-review-boldly-embracing-a-state-of-confusion\">Fontaines DC</a> triumphing (respectively: best new artist and best group; best international group) constituted an upset, but that feels like clutching at straws. Given the nature of the Brits, it’s certainly a little surprising that Fontaines beat K-pop titans <a href=\"https://www.theguardian.com/music/2022/sep/14/blackpink-k-pop-girl-group-born-pink-rule-music-world\">Blackpink</a>, but they’re hardly left-field unknowns: their last album entered the charts at No 1.</p>"
                        }
                    },
                    {
                        "type": "image",
                        "assets": [
                            {
                                "type": "image",
                                "mimeType": "image/jpeg",
                                "file": "https://media.guim.co.uk/b472f2013d29cc88d5d6514e61b65ae7ff6542ba/0_203_5568_3341/5568.jpg",
                                "typeData": {
                                    "aspectRatio": "5:3",
                                    "width": 5568,
                                    "height": 3341
                                }
                            },
                            {
                                "type": "image",
                                "mimeType": "image/jpeg",
                                "file": "https://media.guim.co.uk/b472f2013d29cc88d5d6514e61b65ae7ff6542ba/0_203_5568_3341/master/5568.jpg",
                                "typeData": {
                                    "aspectRatio": "5:3",
                                    "width": 5568,
                                    "height": 3341,
                                    "isMaster": true
                                }
                            },
                            {
                                "type": "image",
                                "mimeType": "image/jpeg",
                                "file": "https://media.guim.co.uk/b472f2013d29cc88d5d6514e61b65ae7ff6542ba/0_203_5568_3341/2000.jpg",
                                "typeData": {
                                    "aspectRatio": "5:3",
                                    "width": 2000,
                                    "height": 1200
                                }
                            },
                            {
                                "type": "image",
                                "mimeType": "image/jpeg",
                                "file": "https://media.guim.co.uk/b472f2013d29cc88d5d6514e61b65ae7ff6542ba/0_203_5568_3341/1000.jpg",
                                "typeData": {
                                    "aspectRatio": "5:3",
                                    "width": 1000,
                                    "height": 600
                                }
                            },
                            {
                                "type": "image",
                                "mimeType": "image/jpeg",
                                "file": "https://media.guim.co.uk/b472f2013d29cc88d5d6514e61b65ae7ff6542ba/0_203_5568_3341/500.jpg",
                                "typeData": {
                                    "aspectRatio": "5:3",
                                    "width": 500,
                                    "height": 300
                                }
                            },
                            {
                                "type": "image",
                                "mimeType": "image/jpeg",
                                "file": "https://media.guim.co.uk/b472f2013d29cc88d5d6514e61b65ae7ff6542ba/0_203_5568_3341/140.jpg",
                                "typeData": {
                                    "aspectRatio": "5:3",
                                    "width": 140,
                                    "height": 84
                                }
                            }
                        ],
                        "imageTypeData": {
                            "caption": "Aitch beat Stormzy to best hip-hop/grime/rap act.",
                            "displayCredit": true,
                            "credit": "Photograph: David Fisher/Global/REX/Shutterstock",
                            "source": "Global/REX/Shutterstock",
                            "photographer": "David Fisher",
                            "alt": "Pop-facing Aitch beat Stormzy to best hip-hop/grime/rap act",
                            "mediaId": "b472f2013d29cc88d5d6514e61b65ae7ff6542ba",
                            "mediaApiUri": "https://api.media.gutools.co.uk/images/b472f2013d29cc88d5d6514e61b65ae7ff6542ba",
                            "suppliersReference": "13660983x",
                            "imageType": "Photograph"
                        }
                    },
                    {
                        "type": "text",
                        "assets": [

                        ],
                        "textTypeData": {
                            "html": "<p>Save some shonky hosting from Mo Gilligan – introducing Lewis Capaldi as Sam Capaldi, and invariably relying on “gags” about alcohol consumption in between awards and performances – and a technical hitch that required showing a 2022 performance from Adele, pretty much everything went to plan. Except, one suspects, solving the Brits’ most obvious problem. We live in an age when pop’s primary audience of teens and young twentysomethings simply aren’t going to sit down and watch a two-and-a-bit-hour award ceremony on live TV: viewing doesn’t work like that any more.</p> \n<p>How moving the Brits to a Saturday was supposed to lure its target market away from social media and on-demand streaming – or indeed just going out with their mates – is an intriguing question. You rather got the feeling that most of this years’ viewers might have been old enough to remember the days when the Brits seemed to give <a href=\"https://www.theguardian.com/music/2022/apr/21/annie-lennox-30-greatest-songs-ranked\">Annie Lennox</a> an award every year, and that they might have spent the evening asking “who’s that?” whenever 2023’s stars appeared. Which can’t possibly be the Brits’ desired effect.</p>"
                        }
                    }
                ]
            }
        ],
        "totalBodyBlocks": 1
    },
    "isHosted": false,
    "pillarId": "pillar/arts",
    "pillarName": "Arts"
}