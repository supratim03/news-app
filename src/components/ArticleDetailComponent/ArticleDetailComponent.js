import React from "react";
import PropTypes from 'prop-types'
import { ArticleBlock, ArticleBody, ArticleContainer, BodyBlock, Headline, HeadlineBlock, PublicationDateBlock, TextBlock, Thumbnail, Title } from "./ArticleDetailStyled";
const ArticleDetailComponent = ({ articleDetail }) => {

    const getTextBodies = articleDetail?.blocks && articleDetail?.blocks?.body.map((text, index) => {
        return (
            <div key={text.id}>
                <p>{text.bodyTextSummary}</p>
            </div>

        )
    })
    return (
        <ArticleContainer>
            <PublicationDateBlock>
                <span>{articleDetail?.webPublicationDate}</span>
            </PublicationDateBlock>
            <ArticleBlock>
                <Title>
                    {articleDetail?.webTitle}
                </Title>
            </ArticleBlock>
            <HeadlineBlock>
                <Headline>
                    {articleDetail?.fields?.headline}
                </Headline>
            </HeadlineBlock>
            <hr />
            <ArticleBody>
                <BodyBlock>
                    <TextBlock>
                        {getTextBodies}
                    </TextBlock>
                    
                    {articleDetail?.fields?.thumbnail && <div><Thumbnail imgUrl={articleDetail?.fields?.thumbnail} /></div>}
                </BodyBlock>
                
            </ArticleBody>
        </ArticleContainer>
    )
}

ArticleDetailComponent.propTypes = {
    articleDetail: PropTypes.objectOf(PropTypes.any).isRequired
}

export default ArticleDetailComponent;