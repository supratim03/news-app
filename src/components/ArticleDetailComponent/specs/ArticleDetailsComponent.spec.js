import React from "react";
import { mount } from "enzyme";
import ArticleDetailComponent from "../ArticleDetailComponent";
import {articleDetail} from '../../../testData/testData'

describe("ArticleDetail component", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <ArticleDetailComponent 
                articleDetail={null}
            />
        );
    });

    it("renders correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
});

describe("ArticleDetail component with value", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <ArticleDetailComponent 
                articleDetail={articleDetail}
            />
        );
    });

    it("renders correctly with data", () => {
        expect(wrapper).toMatchSnapshot();
    });
});