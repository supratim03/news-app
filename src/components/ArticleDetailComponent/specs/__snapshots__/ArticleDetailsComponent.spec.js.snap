// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDetail component renders correctly 1`] = `ReactWrapper {}`;

exports[`ArticleDetail component with value renders correctly with data 1`] = `ReactWrapper {}`;
